src/centralserver/registration-service/build.gradle

```diff
@@ -1,30 +1,84 @@
 plugins {
     id 'io.spring.dependency-management'
     id 'org.springframework.boot'
+    id 'org.openapi.generator' version "$openApiGeneratorVersion"
 }
 
 bootJar {
     manifest {
         attributes(
-            'Implementation-Title'   : 'xroad-centralserver',
-            'Implementation-Version' : xroadVersion
+            'Implementation-Title': 'xroad-centralserver-registration-service',
+            'Implementation-Version': xroadVersion
         )
     }
 }
 
+sourceSets {
+    main {
+        java.srcDirs = [
+            'src/main/java',
+            'build/generated-sources/openapi/src/main/java',
+        ]
+    }
+}
+
+archivesBaseName = 'centralserver-registration-service'
+
+openApiGenerate {
+    generatorName = "spring"
+    inputSpec = "$projectDir/../admin-service/src/main/resources/openapi-definition.yaml"
+    outputDir = "$buildDir/generated-sources/openapi"
+    modelPackage = "org.niis.xroad.centralserver.registrationservice.openapi.model"
+    globalProperties = [
+        apiDocs                : "false",
+        apiTests               : "false",
+        modelDocs              : "false",
+        modelTests             : "false",
+        models                 : "",
+        apis                   : "false",
+        generateSupportingFiles: "false"
+    ]
+    configOptions = [
+        openApiNullable: "false",
+        "swaggerAnnotations": "false"
+    ]
+}
+
+// Fix up to date check
+tasks.openApiGenerate {
+    inputs.files(openApiGenerate.inputSpec)
+    outputs.dir(openApiGenerate.outputDir)
+}
+
+compileJava.dependsOn tasks.openApiGenerate
+
 dependencies {
-    implementation('org.springframework.boot:spring-boot-starter-web') {
-        exclude module: 'spring-webmvc'
-        exclude module: 'spring-boot-starter-json'
+    annotationProcessor "org.springframework:spring-context-indexer"
+    annotationProcessor "org.springframework.boot:spring-boot-configuration-processor"
+
+    implementation(project(':common-util')) {
+        exclude group: 'com.typesafe.akka'
+        exclude group: 'org.eclipse.jetty'
+        exclude group: 'org.quartz-scheduler'
+    }
+
+    implementation(project(':common-verifier')) {
+        transitive=false
     }
 
-    implementation 'com.sun.activation:javax.activation:1.2.0'
-    implementation "com.sun.xml.ws:jaxws-rt:$jaxbVersion"
+    implementation('org.springframework.boot:spring-boot-starter-web') {
+        exclude group: 'org.springframework.boot', module: 'spring-boot-starter-tomcat'
+    }
+    implementation('org.springframework.boot:spring-boot-starter-jetty')
```

Comment:
Here the main reason is to minimize dependencies and Jetty is already a dependency because common-util pulls it in (and it can not be excluded without refactoring common-util module). Perhaps admin-service could use jetty as well.